namespace/consts.go

```diff
@@ -9,6 +9,10 @@ const (
 	// NamespaceVersionSize is the size of a namespace version in bytes.
 	NamespaceVersionSize = 1
 
+	// VersionIndex is the index of the version in the namespace. This should
+	// always be the first byte
+	VersionIndex = NamespaceVersionSize - 1 // 0
```

Comment:
nit: I don't think we need to define this constant in terms of `NamespaceVersionSize` because if we updated `NamespaceVersionSize` to 2 then this constant wouldn't make sense any more so having it auto update to `2 -1 = 1` isn't necessary or desirable. In other words, I think this could be:

```suggestion
	VersionIndex = 0
```